cmd/kube-controller-manager/app/core.go

```diff
@@ -71,9 +71,23 @@ func startServiceController(ctx ControllerContext) (bool, error) {
 }
 
 func startNodeController(ctx ControllerContext) (bool, error) {
-	_, clusterCIDR, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
-	if err != nil {
-		glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+	var clusterCIDRs []*net.IPNet
+	for _, cidr := range ctx.Options.ClusterCIDRs {
+		_, cc, err := net.ParseCIDR(cidr)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR[0], err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+	}
+	NodeCIDRMaskSizes := ctx.Options.NodeCIDRMaskSizes
+	// If the new style was not found, check the old.
+	if len(clusterCIDRs) == 0 {
+		_, cc, err := net.ParseCIDR(ctx.Options.ClusterCIDR)
+		if err != nil {
+			glog.Warningf("Unsuccessful parsing of cluster CIDR %v: %v", ctx.Options.ClusterCIDR, err)
+		}
+		clusterCIDRs = append(clusterCIDRs, cc)
+		NodeCIDRMaskSizes = append(NodeCIDRMaskSizes, int(ctx.Options.NodeCIDRMaskSize))
```

Comment:
Just point out a corner case, since this is end-user provided information. I would **think** that we'd want to allow either old or new syntax, but not both. Would think that we'd want to do all the validation up front, before doing any processing, to avoid subtle conditions and errors could occur.

Do we have UTs that check the failure cases of user providing invalid combinations of inputs and that they are handled correctly?